Update huawei-sun2000.yaml

Hypo93 opened the pull request:
This PR enhances battery control by deactivating Modbus register 47087 when stopping the charge process (case 1). This allows the inverter/battery to enter deep sleep mode and prevents standby losses.

Fix https://github.com/evcc-io/evcc/issues/17692

Hypo93 committed:
Update huawei-sun2000.yaml

This PR enhances battery control by deactivating Modbus register 47087 when stopping the charge process (case 1). This allows the inverter/battery to enter deep sleep mode and prevents standby losses.

https://github.com/evcc-io/evcc/issues/17692

andig commented on templates/definition/meter/huawei-sun2000.yaml:
```diff
@@ -146,11 +146,11 @@ render: |
     reset: 1 # reset watchdog on normal
     set:
-      source: switch
+      source: sequence
```
erst switch, dann sequence. das muss geschachtelt sein!

Hypo93 replied:
Danke, hatte noch mehr Fehler in der Syntax. Hab dies soeben angepasst…

Hypo93 committed:
Update huawei-sun2000.yaml

Hatte die Struktur erst falsch interpretiert. Erster Test mit den Anpassung sieht positiv aus, dass AC Laden wird wie gewünscht aktiviert und deaktiviert. Heute Nacht erfolgt ein Test, ob der WR auch aus dem Tiefschlaf geholt wird zum laden.

MucheteWO commented:
Hallo. Sorry aber das funktioniert so nicht. Bei mir muss AC Ladenan bleiben wegen einer Kaskade (2.WR) und einem externen Balkonkraftwerk. Seit der Version 131.11 tooglet nun mein Main Wechselrichter an dem die Batterie hängt in einer Dauerschleife zwischen Starten und Standby. Ich gehe vorerst auf die Version 131.10 zurück.
Gruß

mucki12 commented:
@MucheteWO 
Kann das Problem mit weiteren Leistungserzeugern nachvollziehen. Allerdings zieht Huawei ohne Standby tatsächlich deutlich mehr Energie als mit. Da die weiteren Leistungserzeuger ja in der Regel nicht dauerhaft Strom produzieren ist das ursprüngliche Verhalten von Huawei schon suboptimal.

Ich bin mittlerweile dazu übergegangen das AC-Laden (an/aus) über Home Assistant zu steuern (weiterer Erzeuger produziert -> AC Laden an / produziert nicht -> AC Laden aus). Somit kann ich den gewünschten Standby nutzen und die Energie von weiteren Erzeugern landet dennoch im LUNA. Aber kann wie gesagt dein geschildertes Problem nachvollziehen...

SolarPowerEV commented:
Wäre es nicht eine Lösung, wenn man vor dem Laden des Hausakkus vom Grid das Register einfach abfragt und dann nach dem Laden wieder auf den Wert setzt, den es vor dem Laden hatte?

Also wenn AC-Laden vorher schon an ist (weil AC-Laden dauerhaft gebraucht wird, wie im Fall von @MucheteWO) dann würde es entsprechend auch am Ende des Ladevorgangs aktiv bleiben.

Und wenn AC-Laden vorher ausgeschaltet war, um den Standby-Eigenverbrauch zu verbessern (wie im Fall von @Hypo93 oder auch in meinem Fall mit 2x SUN2000 WRs und einer LUNA2000), dann würde es auch nach dem Ladevorgang wieder ausgeschaltet.

MucheteWO commented:
> Wäre es nicht eine Lösung, wenn man vor dem Laden des Hausakkus vom Grid das Register einfach abfragt und dann nach dem Laden wieder auf den Wert setzt, den es vor dem Laden hatte?
> 
> Also wenn AC-Laden vorher schon an ist (weil AC-Laden dauerhaft gebraucht wird, wie im Fall von @MucheteWO) dann würde es entsprechend auch am Ende des Ladevorgangs aktiv bleiben.
> 
> Und wenn AC-Laden vorher ausgeschaltet war, um den Standby-Eigenverbrauch zu verbessern (wie im Fall von @Hypo93 oder auch in meinem Fall mit 2x SUN2000 WRs und einer LUNA2000), dann würde es auch nach dem Ladevorgang wieder ausgeschaltet.

Das wäre natürlich absolut super, wenn das so gehen würde.

mucki12 commented:
Zum Glück kann man ja auf der Template Vorlage jederzeit den LUNA als custom Battery in der evcc.yaml anlegen. Habe gerade erst wieder bemerkt, dass ich den LUNA noch immer so angelegt habe (kam noch aus der Zeit, als das Template das Netzladen nur mit 5000W unterstützt hatte).

mpw96 commented:
@SolarPowerEV, @MucheteWO Ich glaube, Eure Idee geht hier unter. Wollt ihr nicht ein neues Issue dafür anlegen?

andig commented:
> Wäre es nicht eine Lösung

Ich fürchte nein…

SolarPowerEV commented:
Warum nicht? Weil man das Register nicht auslesen kann? Oder weil das evcc-Projekt oder Go keine ausgelesenen Register zwischenspeichern können?

Hypo93 commented:
Ich glaube es ist vom Projekt generell nicht gewollt Daten auszulesen und flüchtig zwischen zu speichern.

mucki12 commented:
@snookerap 
Was spricht aus deiner Sicht dagegen den neuen Parameter forceaccharging: entsprechend zu nutzen um genau das alter Verhalten wieder herzustellen?

mucki12 commented:
> Denke aber trotzdem, dass das "Netzladen" immer im Standard AN sein sollte

Schwierig zu beurteilen. Wenn ich raten müsste, dann würde ich mal schätzen, dass wohl >80% der Huawei Besitzer keine Kaskade haben und somit über den Sleepmodus ganz glücklich sein sollten.

Ich persönlich steuere das AC-Laden aber über ein Drittsystem nach einer einfachen Logik (fremder Leistungserzeuger erzeugt Strom -> AC Laden an else AC Laden aus)

SolarPowerEV commented:
> > Denke aber trotzdem, dass das "Netzladen" immer im Standard AN sein sollte
> 
> Schwierig zu beurteilen. Wenn ich raten müsste, dann würde ich mal schätzen, dass wohl >80% der Huawei Besitzer keine Kaskade haben und somit über den Sleepmodus ganz glücklich sein sollten.

Das sehe ich ähnlich. Ich habe sogar eine Kaskade. Dennoch möchte ich im Winter das AC-Laden die meiste Zeit ausgeschaltet haben! Eben weil aufgrund des geringen PV-Ertrags bei nahezu dauerhaft leerem Akku der Wechselrichter unnötig im Standby bleibt und ordentlich Strom zieht. Daher liegt der Prozentsatz vermutlich sogar noch höher, als Du vermutest, @mucki12 .

Dass ich mir nur für dieses Setting eine Bastellösung mit einem externen System bauen müsste, fände ich nicht sonderlich attraktiv.

Mit dem Parameter forceaccharging: kann denjenigen, die den WR unbedingt 24/7 aktiv haben wollen, ja auch leicht geholfen werden.

Schade finde ich nur, dass mein oben gemachter Vorschlag, _vor dem Laden des Hausakkus vom Grid das Register einfach abzufragen und dann nach dem Laden wieder auf den Wert zu setzen, den es vor dem Laden hatte_, nicht umgesetzt werden kann aus Gründen, die hier niemand verraten zu wollen scheint.

Denn das wäre die aus Benutzersicht intuitivste Lösung und evcc würde sich damit am wenigsten "invasiv" verhalten.

mucki12 commented:
> Mit dem Parameter forceaccharging: kann denjenigen, die den WR unbedingt 24/7 aktiv haben wollen, ja auch leicht geholfen werden.

Jeder Jeck ist halt anders :-)
Bei mir ist das keine "Bastellösung" sondern das absolute Optimum. Mein fremder "Wechselrichter" ist eine Brennstoffzelle und die produziert auch gerne mal in der Nacht Strom. Somit wird bei mir immer dann exakt das AC Laden angeschaltet, wenn es wirklich benötigt wird. Wenn die Brennstoffzelle mal keinen Bock auf Produktion hat, dann wird AC Laden ausgeschaltet und die Huawei Kombi darf bei Bedarf einschlafen.

SolarPowerEV commented:
Ich glaube Du hast meine Verwendung von "Bastellösung" missverstanden.
In vielen Fällen macht so eine Steuerung absolut Sinn. In Deinem offensichtlich auch.

Aber ich - wie viele andere Menschen, die sich nicht bereits schon tief in das Thema "Home Automation" reingegraben haben - habe weder die Zeit noch den Nerv, mir eine Lösung basteln zu müssen, damit der Wechselrichter nachts wieder ausgeht, nur weil in evcc das Handling des AC-Laden-Settings so geändert wurde, wie @snookerap es bevorzugen würde.
